cfgmgr/teammgr.cpp

```diff
@@ -0,0 +1,467 @@
+
+        auto alias = kfvKey(t);
+        auto op = kfvOp(t);
+
+        if (op == SET_COMMAND)
+        {
+            SWSS_LOG_INFO("Received port %s state update", alias.c_str());
+
+            vector<string> keys;
+            m_cfgLagMemberTable.getKeys(keys);
+
+            for (auto key : keys)
+            {
+                auto tokens = tokenize(key, '|');
+
+                auto lag = tokens[0];
+                auto member = tokens[1];
+
+                // Find the master of the port
+                if (alias == member)
+                {
+                    addLagMember(lag, alias);
+                    break;
+                }
+            }
+        }
+        else if (op == DEL_COMMAND)
+        {
+            SWSS_LOG_INFO("Received port %s state removal", alias.c_str());
+        }
+
+        it = consumer.m_toSync.erase(it);
+    }
+}
+
+bool TeamMgr::setLagAdminStatus(const string &alias, const string &admin_status)
+{
+    SWSS_LOG_ENTER();
+
+    stringstream cmd;
+    string res;
+
+    // ip link set dev <port_channel_name> [up|down]
+    cmd << IP_CMD << " link set dev " << alias << " " << admin_status;
+    EXEC_WITH_ERROR_THROW(cmd.str(), res);
+
+    SWSS_LOG_NOTICE("Set port channel %s admin status to %s",
+            alias.c_str(), admin_status.c_str());
+
+    return true;
+}
+
+bool TeamMgr::setLagMtu(const string &alias, const string &mtu)
+{
+    SWSS_LOG_ENTER();
+
+    stringstream cmd;
+    string res;
+
+    // ip link set dev <port_channel_name> mtu <mtu_value>
+    cmd << IP_CMD << " link set dev " << alias << " mtu " << mtu;
+    EXEC_WITH_ERROR_THROW(cmd.str(), res);
+
+    vector<FieldValueTuple> fvs;
+    FieldValueTuple fv("mtu", mtu);
+    fvs.push_back(fv);
+    m_appLagTable.set(alias, fvs);
+
+    vector<string> keys;
+    m_cfgLagMemberTable.getKeys(keys);
+
+    for (auto key : keys)
+    {
+        auto tokens = tokenize(key, '|');
+        auto lag = tokens[0];
+        auto member = tokens[1];
+
+        if (alias == lag)
+        {
+            m_appPortTable.set(member, fvs);
+        }
+    }
+
+    SWSS_LOG_NOTICE("Set port channel %s MTU to %s",
+            alias.c_str(), mtu.c_str());
+
+    return true;
+}
+
+bool TeamMgr::addLag(const string &alias, int min_links, bool fallback)
+{
+    SWSS_LOG_ENTER();
+
+    stringstream cmd;
+    string res;
+
+    stringstream conf;
+    conf << "'{\"device\":\"" << alias << "\","
+         << "\"hwaddr\":\"" << m_mac.to_string() << "\","
+         << "\"runner\":{"
+         << "\"active\":\"true\","
+         << "\"name\":\"lacp\"";
+
+    if (min_links != 0)
+    {
+        conf << ",\"min_ports\":" << min_links;
+    }
+
+    if (fallback)
+    {
+        conf << ",\"fallback\":\"true\"";
+    }
+
+    conf << "}}'";
+
+    SWSS_LOG_INFO("Port channel %s teamd configuration: %s",
+            alias.c_str(), conf.str().c_str());
+
+    cmd << TEAMD_CMD << " -r -t " << alias << " -c " << conf.str() << " -d";
```

Comment:
here, you force team device recreation in case if already exist, we probably need to revisit in the warm restart context.

Reply:
sure